aten/src/ATen/native/CPUFallback.cpp

```diff
@@ -0,0 +1,141 @@
+  const auto arguments_begin = stack->size() - num_arguments;
+
+  std::vector<at::Tensor> tensor_args;
+  std::vector<int> tensor_args_indices;
+
+  // Step 1: Convert all non-CPU tensor inputs into CPU tensors
+  // and put them on the stack at the correct indices.
+  for (int64_t idx = 0; idx < arguments.size(); ++idx) {
+    const auto& ivalue = arguments[idx];
+    if (ivalue.isTensor()) {
+      tensor_args.push_back(ivalue.toTensor());
+      tensor_args_indices.push_back(idx);
+    } else if (ivalue.isTensorList()) {
+      // I'm worried about the ivalue storing a reference to the std::vector temporary that I create.
+      // To be extra safe I'm converting to a c10::List, IValue will take ownership of the items in the list.
+      auto cpu_ivalue = c10::IValue(c10::List<at::Tensor>(to_cpu(ivalue.toTensorList().vec())));
+      (*stack)[arguments_begin + idx] = std::move(cpu_ivalue);
+    }
+  }
+  auto cpu_tensors = to_cpu(tensor_args);
+
+  for (auto i = 0; i < tensor_args_indices.size(); ++i) {
+    //auto cpu_ivalue = c10::IValue(cpu_tensors[i]);
+    auto idx = tensor_args_indices[i];
+    //(*stack)[arguments_begin + idx] = std::move(cpu_ivalue);
+    (*stack)[arguments_begin + idx] = c10::IValue(cpu_tensors[i]);
+  }
+
+  // Step 2: Call the underlying CPU implementation of the operator
+  op.redispatchBoxed(c10::DispatchKeySet(c10::DispatchKey::CPU), stack);
+
+  // Step 3: We need to take special care to handle mutable aliases properly:
+  // If any input tensors are mutable aliases, we need to
+  // directly copy the updated data on the CPU tensors back to the original inputs.
+  for (int64_t i = 0; i < tensor_args_indices.size(); ++i) {
+    auto tensor_idx = tensor_args_indices[i];
+    const auto& alias_info = schema_args[tensor_idx].alias_info();
+    if (alias_info.has_value() && alias_info.value().isWrite()) {
+      at::_copy_from_and_resize(cpu_tensors[i], tensor_args[i]);
+    }
+  }
+
+  // Step 4: Convert any CPU output tensors back to the original input device.
+  // For mutable alias'd outputs, we also need to take special care
+  // to move the ORIGINAL input tensor back onto the stack, in place of
+  // the temporary CPU output tensor that we created.
+  //
+  // Also note that we are incapable of handling immutable alises properly.
+  // Why?
+  // Schemas with an immutable alias'd tensor outputs correspond to view operators.
+  // For example, the `view_as` schema from native_functions.yaml:
+  // `view_as(Tensor(a) self, Tensor other) -> Tensor(a)`
+  // We can't handle these ops properly, because view ops are supposed to return
+  // a NEW tensor that shares the SAME storage as the original tensor.
+  // However, the new tensor that we created cannot share the same storage,
+  // since it lives on CPU and the original tensor lives on a different device.
+  // Because of that, we treat immutable aliases the same way that we treat non-aliases:
+  // as a fresh tensor that has entirely new storage.
```

Comment:
Wouldn't it be better to error in this case?

Reply:
Yeah, agreed; I left it like that to maintain the current behavior for XLA, but it's a bug that we should probably error on instead.

The only view op that it applies to for xla is `unfold`, since it's non-composite and doesn't have an XLA lowering:

```
>>> a = torch.arange(4)
>>> b = a.unfold(0, 2, 1)
>>> b[0] = -1
>>> a
tensor([-1, -1,  2,  3])
>>> a = torch.arange(4, device=xm.xla_device())
>>> b = a.unfold(0, 2, 1)
>>> b[0] = -1
>>> a
tensor([0, 1, 2, 3], device='xla:0')
```

Reply:
Yup we should make it clear that XLA need to handle all view ops in the backend instead of relying on fallback. IIRC @JackCaoG tried lowering `unfold` before but it's not merged due to complexity and priority. 
(Erroring out here is definitely helpful).

Reply:
Haven't updated the PR yet, but I just tested the error message locally:
```
>>> a = torch.arange(4, device=xm.xla_device())
>>> b = a.unfold(0, 2, 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: The operator aten::unfold appears to be a view operator, but it has no implementation for the backend "xla:0". View operators don't support falling back to run on the CPU, since the tensor's storage cannot be shared across devices.
```